setup.py

```diff
@@ -1,45 +1,81 @@
-from __future__ import (absolute_import, division, print_function)
-import setuptools
-from distutils.core import setup, Extension
+from __future__ import absolute_import, division, print_function
+
+import sys
+from distutils.core import Extension, setup
+from os import path
+
 import numpy as np
+import setuptools
+
 import versioneer
 
-with open('requirements.txt') as f:
+min_version = (3, 8)
+if sys.version_info < min_version:
+    error = """
+bluesky-adaptive does not support Python {0}.{1}.
+Python {2}.{3} and above is required. Check your Python version like so:
+
+python3 --version
+
+This may be due to an out-of-date pip. Make sure you have pip >= 9.0.1.
+Upgrade pip like so:
+
+pip install --upgrade pip
+""".format(
+        *(sys.version_info[:2] + min_version)
+    )
+    sys.exit(error)
+
+here = path.abspath(path.dirname(__file__))
+
+with open(path.join(here, "README.md"), encoding="utf-8") as readme_file:
```

Comment:
The markdown format requires an extra kwarg, see https://github.com/NSLS-II/scientific-python-cookiecutter/issues/25.